File: src/main/java/com/example/JettyRunner.java
package com.example;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.ServletException;
 
import java.io.IOException;
 
import org.eclipse.jetty.server.Server;
import org.eclipse.jetty.server.Request;
import org.eclipse.jetty.server.handler.AbstractHandler;
 
public class JettyRunner extends AbstractHandler
{
    private String message = "Hello World";

    public JettyRunner(String message){
        super();
        this.message = message;
    }

    public void handle(String target,
                       Request baseRequest,
                       HttpServletRequest request,
                       HttpServletResponse response) 
        throws IOException, ServletException
    {
        response.setContentType("text/html;charset=utf-8");
        response.setStatus(HttpServletResponse.SC_OK);
        baseRequest.setHandled(true);
        response.getWriter().println(message);
    }
 	
 	public static void run(int port, int xMilliSeconds, String message) throws Exception{
 	    Server server = new Server(port);
        server.setHandler(new JettyRunner(message));
        server.start();
        Thread.sleep(xMilliSeconds);
        server.stop();
        server.join();	
 	}
}

File: src/main/scala/com/example/JettyRunner.scala
package com.example

import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
import org.eclipse.jetty.server.{Request, Server}
import org.eclipse.jetty.server.handler.AbstractHandler

object FailedRunner extends App {
  def main(): Unit = {
    print("Something went wrong...")
  }
}

class MyServer(port: Int, message: String) {
  val server = new Server(port)
  server.setHandler(new JettyRunner(message))

  def start(): Unit = {
    server.start()
    //throw new RuntimeException("I cannot work!!")
    Thread.sleep(5000)
  }

  def stop():Unit = {
    server.stop()
    server.join()
  }


}

class JettyRunner(message: String = "hello-world") extends AbstractHandler {
  override def handle(target: String, baseRequest: Request, request: HttpServletRequest, response: HttpServletResponse): Unit = {
    response.setContentType("text/html;charset=utf-8")
    response.setStatus(HttpServletResponse.SC_OK)
    baseRequest.setHandled(true)
    response.getWriter.println(message)
  }
}